api/src/main/java/com/exelaration/abstractmemery/controllers/UserController.java

```diff
@@ -0,0 +1,27 @@
+package com.exelaration.abstractmemery.controllers;
+
+import com.exelaration.abstractmemery.domains.ApplicationUser;
+import com.exelaration.abstractmemery.repositories.UserRepository;
+import com.exelaration.abstractmemery.services.implementations.UserDetailsServiceImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/users")
+public class UserController {
+
+  @Autowired private UserDetailsServiceImpl userDetailsService;
+  @Autowired private UserRepository userRepository;
+
+  @PostMapping("/sign-up")
+  public String signUp(@RequestBody ApplicationUser user) {
+    if (userRepository.findByUsername(user.getUsername()) == null) {
```

Comment:
This should be done by the `userService`.  Something like `if (!userService.userExists(user.getUsername()))`